Modules/Sources/WordPressUI/Views/Acknowledgements/AcknowledgementsListViewModel.swift

```diff
@@ -0,0 +1,55 @@
+import SwiftUI
+
+open class AcknowledgementsListViewModel: ObservableObject {
+
+    @Published
+    public var items: [AcknowledgementItem]
+
+    @Published
+    public var isLoadingItems: Bool
+
+    @Published
+    public var error: Error?
+
+    private var taskHandle: Task<Void, Error>?
+
+    open func loadItems() async throws -> [AcknowledgementItem] {
+        abort() // This should never run – subclass this ViewModel and override this method
+    }
+
+    public init() {
+        self.items = []
+        self.isLoadingItems = true
+    }
+
+    private init(items: [AcknowledgementItem]) {
+        self.items = items
+        self.isLoadingItems = false
+    }
+
+    @MainActor
+    func onAppear() {
+        self.taskHandle = Task {
+            do {
+                let items = try await self.loadItems()
```

Comment:
This task inherits actor from the function, so it already runs on MainActor. If you want to make it explicit, I suggest using `Task { @MainActor in`.
 
The `loadItems() async` method, as a standalone async method, will always run in the background by default, so it will be fine. `AcknowledgementsService` is also itself an actor, so it will also guarantee to run in the background.

Reply:
Should be sorted by feb005a028c3d80d3efe9fd7bed5a9b44fec15f2